crates/wast/src/core.rs

```diff
@@ -14,8 +14,14 @@ pub fn val<T>(store: &mut Store<T>, v: &WastArgCore<'_>) -> Result<Val> {
         F32(x) => Val::F32(x.bits),
         F64(x) => Val::F64(x.bits),
         V128(x) => Val::V128(u128::from_le_bytes(x.to_le_bytes()).into()),
-        RefNull(HeapType::Extern) => Val::ExternRef(None),
-        RefNull(HeapType::Func) => Val::FuncRef(None),
+        RefNull(HeapType::Abstract {
+            ty: AbstractHeapType::Extern,
+            ..
```

Comment:
Mind matching `shared: false` here explicitly and panicking/bailing on `shared: true`?